src/ResourceManager/AnalysisServices/Commands.AnalysisServices.ServiceManagement/Models/AsAzureClientSession.cs

```diff
@@ -0,0 +1,142 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
+using System.Security;
+
+namespace Microsoft.Azure.Commands.AnalysisServices.ServiceManagement
+{
+    /// <summary>
+    /// Represents current AS Azure session.
+    /// </summary>
+    public class AsAzureClientSession
+    {
+        public const string RestartEndpointPathFormat = "/webapi/servers/{0}/restart?api-version=2016-10-01";
+        public const string AadAuthorityUrlProd = "https://login.windows.net";
+        public const string AadAuthorityUrlPpe = "https://login.windows-ppe.net";
```

Comment:
You should not encode these here, you should use the values from the current context, and make use of the constants that we have for these.

You should also not encode internal URIs in your cmdletrs.